src/components/Table/Table.tsx

```diff
@@ -0,0 +1,257 @@
+    if (this.props.defaultSortField) {
+      this.sortData(this.props.defaultSortField);
+    }
+  }
+
+  // Get class names for table
+  getTableClassName() {
+    const {
+      bordered,
+      highlight,
+      striped,
+      theme,
+    } = this.props;
+
+    return classNames(
+      theme.table,
+      bordered && theme.bordered,
+      highlight && theme.highlight,
+      striped && theme.striped
+    );
+  }
+
+  /*** 
+    // Function to take childrens of this component & render it with actual values
+    renderChildren() {
+      return React.Children.map(this.props.children, (child: React.ReactElement<any>) => {
+        // Get the name of component, this could be either ThemedTableHeader, ThemedTableBody, ThemedTableFooter
+        const componentName = typeof child.type === 'function' ? child.type.displayName : '';
+
+        // Get table body & iterate through data received in props
+        // This data then should be initialized in specific td's by checking there dataKey
+        if (componentName === 'ThemedTableBody') {
+          // Last argument is the children return after going through the process of creating rows with actual data
+          return React.cloneElement(child, {}, this.renderTableBody(child));
+        }
+
+        // Returning thead as it is
+        return child;
+      });
+    }
+
+    // Function to take tbody & get its children which most probably always will be tr
+    // Here iterate through the data received by Table component & create / clone tr with those data
+    renderTableBody(tableBody: React.ReactElement<any>): any {
+      const { data } = this.props;
+
+      return React.Children.map(tableBody.props.children, (child: React.ReactElement<any>) => {
+        return data.map((item: any) => {
+          // Data here will be like {name: '', email: '', ...}
+          // This will match with the dataKey passed through TableData & replace its value with data.name, data.emai, ...
+          return React.cloneElement(child, {}, this.renderTableRow(item, child));
+        });
+      });
+    }
+
+    // Get the children row & replace the values with actual data
+    // Match data.key with dataKey sent in TableData
+    renderTableRow(jsonRow:any, tableRow: React.ReactElement<any>): any {
+      return React.Children.map(tableRow.props.children, (child: React.ReactElement<any>) => {
+        if (jsonRow[child.props.dataKey]) {
+          return React.cloneElement(child, { dataKey: jsonRow[child.props.dataKey] });
+        }
+
+        return React.cloneElement(child);
+      });
+    }
+ */
+
+ // Render the thead with th & contain specific header label
+ // Used certain flags which will help to add sorting for any specific fields
+  renderHeader() {
+    const { column, sorting } = this.props;
+    const { field, order } = this.state.sort;
+
+    return (
+      <TableHeader>
+        <TableRow>
+          {
+            column.map((item: any) => {
```

Comment:
@dheeraja00  As stated above, we need an explicit interface, so it should be `(item: ColumnConfiguration) => { ... }`